checks/testdata/workflow-non-github-not-pinned.yaml

```diff
@@ -0,0 +1,48 @@
+# Copyright 2021 Security Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
please add additional unit tests:
1. mix of pinned and non-pinned GitHub actions. score should be 8
2. mixed of pinned and non-pinned non-GitHub dependencies. score should be 2
3. mixed of the above. score should be 0
4. all github and 3p actions pinned. score should be 10. For this one, you can simply update an existing workflow instead of creating a new file. Or you can create a new file: up to you.